File: Data Structures & Algorithms/validate-parentheses/submission-2.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        closeToOpen = { ")" : "(", "]" : "[", "}" : "{" }

        for c in s:
            if c in closeToOpen:
                if stack and stack.pop() == closeToOpen[c]:
                    continue
                else:
                    return False
            else:
                stack.append(c)
        
        return True if not stack else False

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-2.java
class Solution {
    public int evalRPN(String[] tokens) {
        
        ArrayDeque<Integer> stack = new ArrayDeque<>();

        int i = 0;
        while (i < tokens.length) {
            if ("+-*/".contains(tokens[i])) {
                Integer left = stack.pop();
                Integer right = stack.pop();
                switch(tokens[i]) {
                    case "+":
                        stack.push(right + left);
                        break;
                    case "-":
                        stack.push(right - left);
                        break;
                    case "*":
                        stack.push(right * left);
                        break;
                    case "/":
                        stack.push(right / left);
                        break;
                }
                

            } else {
                stack.push(Integer.valueOf(tokens[i]));
            }
            i++;
        }
        return stack.pop();
    }
}
